src/app/shared/spinner/spinner.component.ts

```diff
@@ -1,15 +1,21 @@
 import { Component, OnInit } from '@angular/core';
+import { SpinnerService } from './spinner.service';
 
 @Component({
   selector: 'app-spinner',
   templateUrl: './spinner.component.html',
-  styleUrls: ['./spinner.component.css']
+  styleUrls: ['./spinner.component.css'],
+  providers: [SpinnerService],
 })
 export class SpinnerComponent implements OnInit {
+  showSpinner: boolean;
 
-  constructor() { }
+  constructor(private spinnerService: SpinnerService) { }
 
   ngOnInit() {
+    this.spinnerService.status.subscribe((value: boolean) => {
+      this.showSpinner = value;
+    });
```

Comment:
What do you need this logic for?

Reply:
Now this logic is needed because spinner logic is moved from app component to spinner component.